plugins/plugins.go

```diff
@@ -190,3 +185,38 @@ func (m *Manager) Services() []string {
 	}
 	return s
 }
+
+// parseServicesConfig returns a set of named service clients. The service
+// clients can be specified either as an array or as a map. Some systems (e.g.,
+// Helm) do not have proper support for configuration values nested under
+// arrays, so just support both here.
+func parseServicesConfig(raw json.RawMessage) (map[string]rest.Client, error) {
+
+	services := map[string]rest.Client{}
+
+	var arr []json.RawMessage
+	var obj map[string]json.RawMessage
+
+	if err := util.Unmarshal(raw, &arr); err == nil {
+		for _, s := range arr {
+			client, err := rest.New(s)
+			if err != nil {
+				return nil, err
+			}
+			services[client.Service()] = client
+		}
+	} else if err2 := util.Unmarshal(raw, &obj); err2 == nil {
```

Comment:
[nit] since we don't need `err2`, this could be
```suggestion
	} else if util.Unmarshal(raw, &obj) == nil {
```